src/history/hs_rec.c

```diff
@@ -327,8 +326,9 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_RECONCILE *r, WT_MULTI *mult
     insert_cnt = 0;
     WT_TIME_WINDOW_INIT(&tw);
 
-    WT_RET(__wt_curhs_open(session, NULL, &hs_cursor));
-    F_SET(hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
+    if (r->hs_cursor == NULL)
+        WT_RET(__wt_curhs_open(session, NULL, &r->hs_cursor));
+    F_SET(r->hs_cursor, WT_CURSTD_HS_READ_COMMITTED);
```

Comment:
Is it necessary here to call `reset` or any other function to clean any prior state the cursor could have been into?

Reply:
I think it is better to check whether WT_CURSTD_HS_READ_COMMITTED is set or not and clear the flag if it is not set when entering this function.

Reply:
Do the changes in 1c680c96fcd94f00715422f72c88806beb32edfb work?